### DISCUSSION TITLE
How to trace the precision error in the program ?

### DISCUSSION THREAD

**Discussion by Kim1230** [verificarlo/verificarlo#325] [Q&A] [answered]
Hello, I use verificarlo to test a program which has function A and function B. While the output of A is the input of B. I want to test the effect of  A's result on B's result. So, I want to use MAC_Backends in the function A and IEEE backends in the function B . So, I can trace the affected operations in B. But how I can accomplish this ?  Hope for your response

**Comment by Kim1230** [comment]
And I also have a question about the mca backends and vprec backends. I can see the mode part to control the arithmetic error. I can't understand it . Such as inbound precision and outbound precision. The error I know for the float precision are rounding error and overflow or underflow. Can you give me some documents to help me understand it ? Thanks!

**Reply by pablooliveira** [reply]
You can have a look at our [tutorial](https://github.com/verificarlo/verificarlo/wiki/Tutorials). For a more detailed explanation, you can have a look at chapters 5 and 6 of www.sifflez.org/publications/hdr/habilitation.pdf.

**Comment by pablooliveira** [comment, marked as answer]
Hi @Kim1230. For your initial use case, the easiest is to only apply MCA backend to function A.

You can accomplish this with the `--function=` parameter which limits instrumentation to a single function. Its usage is documented [here](https://github.com/verificarlo/verificarlo/blob/master/doc/03-inclusion-exclusion.md). For complex scenarios, there are more complex ways to select which parts of the program are instrumented.

When using `--function=` you must ensure that the compiler is not inlining the function, you can achieve this by adding `__attribute__((noinline))` to the function prototype.
